enhancements/microshift/default-csi-plugin.md

```diff
@@ -0,0 +1,343 @@
+
+#### Variation
+
+**Start MicroShift with a user-provided ODF-LVM configuration**
+
+> After the device owner has logged into the device ... 
+
+1. The device owner will create a file called `/etc/microshift/lvmd.yaml`
+1. The device owner will define the lvmd config, according to its 
+[documentation](https://github.com/red-hat-storage/topolvm/blob/main/docs/lvmd.md)
+
+   _For example:_
+   
+    ```shell
+    $ cat <<EOF > /etc/microshift/lvmd.yaml
+    socket-name: 
+    deviceClasses:
+    - default: true
+      name: "user-provided-config"
+      volume-group: rhel
+      spare-gb: 10 # in GB
+    socket-name: /run/lvmd/lvmd.sock # socket-name must always be /run/lvmd/lvmd.sock
+    ```
+
+1. The device owner will start the microshift service, and proceed to validate the platform as described in **Validate 
+ODF-LVM function**
+1. The device owner will validate the system, following workflow **Validate ODF-LVM function**
+
+**Start MicroShift with a *malformed* user-provided ODF-LVM configuration**
+
+> After the device owner has logged into the device ...
+
+1. The device owner creates an incorrect lvmd config file under /etc/microshift/lvmd.config
+
+    _For example:_
+    ```shell
+    [root@ushift ~]# cat <<EOF > /etc/microshift/lvmd.yaml
+    socket-name: 
+    deviceClasses:
+    - default: true
+      name: "user-provided-config"
+      volume-group: non_existent_volume_group
+      spare-gb: 10 # in GB
+    socket-name: /run/lvmd/lvmd.sock
+    ```
+**Correct a _malformed_ user-provided ODF-LVM configuration and restart MicroShift**
+
+> After the device owner has logged into the device ...
+
+1. The device owner will edit the /etc/microshift/lvmd.yaml file to correct the error
+1. The device owner will restart MicroShift: `systemctl restart microshift`
+1. The device owner will validate the system, following workflow **Validate ODF-LVM function**
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### API Impacts
+
+**Configuration API**
+
+This proposal extends the MicroShift config API with the optional lvmd configuration.  The lvmd config is not added to 
+the core MicroShift configuration, and must be written to `/etc/microshift/lvmd.yaml`.  This approach aligns with
+MicroShift's  design goals regarding configuration, defined
+[here](https://github.com/openshift/microshift/blob/main/docs/design.md#configuration).
+
+**Cluster CRDs**
+
+ODF-LVM represents logical volumes with a CRD called `logicalvolumes.topolvm.cybozu.com`.  This CRD is installed on the 
+cluster automatically by MicroShift during startup. 
+
+**Always Override Socket-Name**
+
+The ODF-LVM manifests embedded in MicroShift assume the unix domain socket to be accessible at `/run/lvmd/lvmd.sock` in
+the `topolvm-node` pod's filesystem.  Because of this, `socket-name` values in `/etc/microshift/lvmd.yaml` will always
+be overridden to `/run/lvmd/lvmd.sock`.  Values other than this, whether valid or not, will always be overridden.
```

Comment:
Fair point.  Instead of the entire lvmd object, we could expose only the device-classes array.  That would exclude only the socket-name key from the file.

Reply:
Another way to do it would be to allow the user to override the value and for us to read the lvmd.conf file ourselves to get it.

Reply:
I think we're better off not exposing it at all unless there's a clear reason to do so.

Reply:
Yeah, we want to expose the whole file because we want that to be part of the "OS configuration" fed into MicroShift. The question in my mind is whether we really care about that socket name value, or whether we can just take what the user gives us and use that. I _thought_ we could inject the value into the pods we create. Is that wrong?

Reply:
No, I think you're right that it doesn't matter where in the pod the socket is mounted, so long as that path is passed to the lvmd process via it's cmd line flag.

I think we'd be mostly find letting users set the path, and provide them with clear examples and docs.